src/main/java/seedu/moneygowhere/data/currency/CurrencyManager.java

```diff
@@ -25,10 +26,15 @@ public void addCurrency(Currency currency) {
         exchangeRates.put(currencyCode, rate);
     }
 
-    public boolean hasCurrency(String currency) throws CurrencyInvalidException {
+    public boolean hasCurrency(String currency) throws CurrencyInvalidException, CurrencyRatesNotFoundException {
         if (exchangeRates.containsKey(currency)) {
             return true;
         }
+        if (exchangeRates.isEmpty()) {
+            System.out.println("no way");
+            throw new CurrencyRatesNotFoundException(Messages.CURRENCY_MANAGER_RATES_NOT_FOUND);
+        }
+        System.out.println("lolol");
```

Comment:
Haha may I check is this an intended error message?

Reply:
oh my i was doing some testing and i forgot to remove it. My bad!

Reply:
Hahahaha no problem. Glad we caught it before we merged.